config.go

```diff
@@ -105,6 +106,13 @@ func (c *Config) Discover() error {
 		}
 	}
 
+	if len(c.Providers) == 0 {
+		return errors.New("no providers found")
```

Comment:
You're right. I saw some errors.New calls, but fmt.Errorf is clearly the dominant style.